docs/basics/101-137-history.rst

```diff
@@ -225,8 +225,28 @@ in your dataset -- just not written to the history anymore. Let's
 try this to get a feel for it.
 
 The COMMIT in the command can either be a hash or a reference
-with the HEAD pointer. Let's stay with the hash, and reset to the
-commit prior to saving the Gitjokes.
+with the HEAD pointer.
+
+.. findoutmore:: Git terminology: branches and HEADs?
+
+   A Git repository (and thus any DataLad dataset) is build up as a tree of
+   commits. A *branch* is a named pointer (reference) to a commit, and allows
+   to isolate developments. The default branch is called ``master``. ``HEAD`` is
```

Comment:
A few typos:
"A Git repository (and thus any DataLad dataset) is built up as a tree of commits."

"A *branch* is a named pointer (reference) to a commit, and allows you to isolate developments."